sdk/go/common/workspace/plugins.go

```diff
@@ -1389,10 +1390,54 @@ func GetPluginPath(kind PluginKind, name string, version *semver.Version,
 	return path, err
 }
 
+func attemptToDownloadMissingPlugin(kind PluginKind, name string, version *semver.Version) error {
+	pluginSpec := PluginSpec{
+		Kind: kind,
+		Name: name,
+	}
+
+	if version == nil {
+		latestVersion, err := pluginSpec.GetLatestVersion()
+		if err != nil {
+			return err
+		}
+
+		version = latestVersion
+	}
+
+	pluginSpec.Version = version
+
+	withProgress := func(stream io.ReadCloser, size int64) io.ReadCloser {
+		header := fmt.Sprintf("Downloading plugin %s v%s", pluginSpec.Name, version.String())
+		return ReadCloserProgressBar(stream, size, header, colors.Always)
+	}
+
+	retry := func(err error, attempt int, limit int, delay time.Duration) {
+		cmdutil.Diag().Warningf(
+			diag.Message("", "Error downloading plugin: %s\nWill retry in %v [%d/%d]"), err, delay, attempt, limit)
+	}
+
+	downloadedFile, err := DownloadToFile(pluginSpec, withProgress, retry)
+	if err != nil {
+		return fmt.Errorf("error downloading %s: %w", pluginSpec.Name, err)
+	}
+
+	logging.V(1).Info("installing plugin")
+	return pluginSpec.Install(downloadedFile, false)
+}
+
 func GetPluginInfo(kind PluginKind, name string, version *semver.Version,
 	projectPlugins []ProjectPlugin) (*PluginInfo, error) {
 	info, path, err := getPluginInfoAndPath(kind, name, version, false, projectPlugins)
 	if err != nil {
+		if _, ok := err.(*MissingError); ok {
```

Comment:
getPluginInfoAndPath is doing this download, no need to do it here as well.

Reply:
This is needed for #10216  because `getPluginInfoAndPath` only downloads `if !enableLegacyPluginBehavior && version != nil`. For `pulumi convert` the `version` is `nil` when there are no resource plugins available